### PR TITLE
[Rector] Apply Rector: VarConstantCommentRector

### PR DESCRIPTION
**Pull request by samsonasik**
Add `@var` to Class Constants with `VarConstantCommentRector` rector rule.

**Checklist:**
- [x] Securely signed commits

### CODE REVIEW

File: system/HTTP/URI.php [hunk 1]
```diff
@@ -23,13 +23,15 @@ class URI
 	 * Sub-delimiters used in query strings and fragments.
 	 *
 	 * @const string
```
Comment (paulbalandan) [inline system/HTTP/URI.php:25]: Not related but is `@const` a valid phpdoc?

**Review by paulbalandan — APPROVED** [review]

File: system/HTTP/URI.php (hunk 1 above)
Comment (samsonasik) [reply system/HTTP/URI.php:25]: I can't find the reference for `@const` in php documentor manual, remove it?

Comment (paulbalandan) [reply system/HTTP/URI.php:25]: That'd be nice. 😁

Comment (samsonasik) [reply system/HTTP/URI.php]: removed https://github.com/codeigniter4/CodeIgniter4/pull/4759/commits/52a1823e589f6a9852af4efad180e8489e09403c 👍

Comment (paulbalandan) [reply system/HTTP/URI.php]: Thanks

**Comment by samsonasik** [comment]
@paulbalandan thank you for the review.

**Comment by MGatner** [comment]
I realize I missed the boat on this, but this seems unnecessary. Class constants are constant, so their type is always apparent from their declaration. What does adding docblocks accomplish?

**Comment by samsonasik** [comment]
@MGatner that mostly for consistency with other constants that has `@var`

**Comment by MGatner** [comment]
I would rather see a rule that pulled all `@var` tags from constants. "var" is shorthand for "variable" (literally "values that change") and is decidedly not "constant". In addition to being unnecessary I suspect this is probably a docblock violation (though I haven't researched that).

**Comment by samsonasik** [comment]
This is example in phpDocumentor  for using `@var` for const.

https://github.com/phpDocumentor/phpDocumentor/blob/afecb1227c943f31633ef4587769fbdfa7e47cc7/tests/data/ApiTagTest.php#L65-L74

I will try create new custom rule for pull `@var` for Class Constants if we have decision is to remove them.

**Comment by MGatner** [comment]
I do see that constants can have `@var` tags (also found this https://github.com/phpDocumentor/phpDocumentor/blob/dbbe6c48e3553870f3077d8dcc2eda57a1df2c3b/src/phpDocumentor/Descriptor/ConstantDescriptor.php#L109-L123). But does that mean they should? All I can find on that topic is debates on StackOverflow 😅

I'd say at this point it is up to us. I won't fight it if you all want them there, doesn't hurt anything.

**Comment by samsonasik** [comment]
@MGatner ok, here PR to remove it https://github.com/codeigniter4/CodeIgniter4/pull/4766

**Pull request merged**